src/main/scala/com/campudus/tableaux/HelperFunctions.scala

```diff
@@ -11,97 +12,108 @@ object HelperFunctions {
-    tryMap((x: Seq[Int]) => x.map(_.longValue()), InvalidJsonException(s"Warning: ColumnIds should be Numbers", "invalid"))(seq)
-  }
-
   private def checkNotNullArray(json: JsonObject, field: String): ArgumentCheck[JsonArray] = notNull(json.getArray(field), field)
 
-  private def checkForString(seq: Seq[String]): ArgumentCheck[Seq[String]] = {
-    tryMap((y: Seq[String]) => y map { x: String => x }, InvalidJsonException(s"Warning: ColumnNames should be Strings", "invalid"))(seq)
+  private def checkForJsonObject(seq: Seq[JsonObject]): ArgumentCheck[Seq[JsonObject]] = {
+    tryMap((y: Seq[JsonObject]) => y map { x: JsonObject => x }, InvalidJsonException(s"Warning: Columns should be in JsonObjects", "object"))(seq)
   }
 
-  def toTableauxType(list: Seq[String]): ArgumentCheck[Seq[TableauxDbType]] = {
-    sequence(list map (tryMap(Mapper.getDatabaseType, InvalidJsonException("Warning: No such type", "type"))))
+  def toTableauxType(kind: String): ArgumentCheck[TableauxDbType] = {
+    tryMap(Mapper.getDatabaseType, InvalidJsonException("Warning: No such type", "type"))(kind)
   }
 
-  def jsonToSeqOfColumnNameAndType(json: JsonObject): Seq[(String, TableauxDbType)] = {
-    (for {
-      names <- checkNotNullArray(json, "columnName")
-      types <- checkNotNullArray(json, "type")
-      namesAsStringList <- asCastedList[String](names)
-      typesAsStringList <- asCastedList[String](types)
-      namesList <- nonEmpty(namesAsStringList, "columnName")
-      checkedNameList <- checkForString(namesList)
-      typesList <- nonEmpty(typesAsStringList, "type")
-      realTypes <- toTableauxType(typesList)
-      zippedList <- checkSameLengthsAndZip(checkedNameList, realTypes)
-    } yield zippedList).get
-  }
-
-  def jsonToSeqOfRowsWithColumnIdAndValue(json: JsonObject): Seq[Seq[(Long, _)]] = {
-    (for {
-      columnIds <- checkNotNullArray(json, "columnIds")
-      values <- checkNotNullArray(json, "values")
-      namesAsJsonArrayList <- asCastedList[JsonArray](columnIds)
-      typesAsJsonArrayList <- asCastedList[JsonArray](values)
-      columnIdsList <- nonEmpty(namesAsJsonArrayList, "columnIds")
-      valuesList <- nonEmpty(typesAsJsonArrayList, "values")
-      zippedList <- checkSameLengthsAndZip(columnIdsList, valuesList)
-      result <- checkSeq(zippedList)
-    } yield result).get
-  }
-
-  private def checkSeq(seq: Seq[(JsonArray, JsonArray)]): ArgumentCheck[Seq[Seq[(Long, _)]]] = {
-    sequence(seq map { x => jsonArraysToSeq(x) })
+  private def getLinkInformation(json: JsonObject): ArgumentCheck[Option[(IdType, IdType, IdType)]] = for {
+    toTable <- notNull(json.getLong("toTable"), "toTable")
+    toColumn <- notNull(json.getLong("toColumn"), "toColumn")
+    fromColumn <- notNull(json.getLong("fromColumn"), "fromColumn")
+  } yield Some(toTable, toColumn, fromColumn)
+
+  private def seqOfJsonObjectsToTuples(seq: Seq[JsonObject]): ArgumentCheck[Seq[(String, TableauxDbType, Option[(IdType, IdType, IdType)])]] = for {

```

Comment:
It would also make sense to change the name of the method to something more meaningful :)